### PR TITLE
RF test: relax the validation restriction for cpu and gpu predictions

### PR DESCRIPTION
**Pull request by wbo4958**

### CODE REVIEW

**Comment by wbo4958** [comment]
build

File: python/tests/test_random_forest.py [hunk 1]
```diff
@@ -359,11 +359,15 @@ def test_random_forest_classifier(
         result = spark_rf_model.transform(test_df).collect()
         pred_result = [row.prediction for row in result]
 
+        # no need to compare all feature types.
         if feature_type == feature_types.vector:
-            # no need to compare all feature type.
             spark_cpu_result = spark_rf_model.cpu().transform(test_df).collect()
             spark_cpu_pred_result = [row.prediction for row in spark_cpu_result]
-            assert array_equal(spark_cpu_pred_result, pred_result)
+            # Since the precision issue, we can ensure all the predictions are same.
```
Comment (leewyang) [inline python/tests/test_random_forest.py:366]: "Since cuML uses FP32, but Spark uses FP64, we cannot ensure that all predictions will be the same."?

Comment (leewyang) [reply python/tests/test_random_forest.py:366]: Otherwise LGTM

**Review by leewyang — APPROVED** [review]

File: python/tests/test_random_forest.py (hunk 1 above)
Comment (wbo4958) [reply python/tests/test_random_forest.py:366]: I will update the comment in the following PR. For now, let's just merge it.

**Pull request merged**